web/src/components/vis/mixins/plotData.js

```diff
@@ -0,0 +1,54 @@
+import { mapState, mapActions, mapMutations } from 'vuex';
+import { LOAD_PLOT } from '../../../store/actions.type';
+import { SET_PLOT } from '../../../store/mutations.type';
+
+/**
+ * Expects `id` to exist on the wrapping component
+ * @param {String} plotName name of the plot to subscribe to
+ */
+export default function (plotName) {
+  return {
+    data() {
+      return {
+        active: true,
+      };
+    },
+    activated() { this.active = true; },
+    deactivated() { this.active = false; },
+    mounted() { this.reloadPlot(); },
+    computed: {
+      ...mapState(['loading']),
+      dataset() { return this.$store.getters.dataset(this.id); },
+      plot() { return this.$store.getters.plot(this.id, plotName); },
+    },
+    watch: {
+      // eslint-disable-next-line func-names
```

Comment:
This is to avoid having to name the anonymous function in the next line, right? Is there any reason not to give it a low-key name like `_` or `x` or something?

Reply:
I had no idea you could name inline functions.